ibm/resource_ibm_ob_logging.go

```diff
@@ -0,0 +1,358 @@
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Discovered agent",
+			},
+
+			obLoggingNamespace: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Namespace",
+			},
+		},
+	}
+}
+
+func resourceIBMLoggingCreate(d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	var ingestionkey string
+	var privateEndpoint bool
+
+	//Read cluster ID and sysdif instanceID
+	clusterName := d.Get(obLoggingCluster).(string)
+	loggingInstanceID := d.Get(obLoggingInstanceID).(string)
+
+	//Read Ingestionkey
+	if iKey, ok := d.GetOk(obLoggingIngestionkey); ok {
+		ingestionkey = iKey.(string)
+	}
+
+	//Read private enpoint
+	if endPoint, ok := d.GetOk(obLoggingPrivateEndpoint); ok {
+		privateEndpoint = endPoint.(bool)
+	}
+
+	//populate sysdig configure create request
+	params := v2.LoggingCreateRequest{
+		Cluster:         clusterName,
+		IngestionKey:    ingestionkey,
+		LoggingInstance: loggingInstanceID,
+		PrivateEndpoint: privateEndpoint,
+	}
+
+	targetEnv, err := getLoggingTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+
+	var logging v2.LoggingCreateResponse
+	err = resource.Retry(10*time.Minute, func() *resource.RetryError {
+		var err error
+		logging, err = client.Logging().CreateLoggingConfig(params, targetEnv)
+		if err != nil {
+			log.Printf("[DEBUG] logging Instance err %s", err)
+			if strings.Contains(err.Error(), "The user doesn't have enough privileges to perform this action") {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		logging, err = client.Logging().CreateLoggingConfig(params, targetEnv)
+	}
+	if err != nil {
+		return fmt.Errorf("error latching logging instance to cluster: %w", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", clusterName, logging.InstanceID))
+
+	return resourceIBMLoggingRead(d, meta)
+}
+
+func getLoggingTargetHeader(d *schema.ResourceData, meta interface{}) (v2.LoggingTargetHeader, error) {
+	_, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return v2.LoggingTargetHeader{}, err
+	}
+
+	userDetails, err := meta.(ClientSession).BluemixUserDetails()
+	if err != nil {
+		return v2.LoggingTargetHeader{}, err
+	}
+	accountID := userDetails.userAccount
+
+	targetEnv := v2.LoggingTargetHeader{
+		AccountID: accountID,
+	}
+
+	return targetEnv, nil
+}
+
+func resourceIBMLoggingRead(d *schema.ResourceData, meta interface{}) error {
+
+	client, err := meta.(ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+
+	parts, err := idParts(d.Id())
+	if err != nil {
+		return err
+	}
+	clusterName := parts[0]
+	loggingID := parts[1]
+
+	targetEnv, err := getLoggingTargetHeader(d, meta)
+	if err != nil {
+		return err
+	}
+
+	config, err := client.Logging().GetLoggingConfig(clusterName, loggingID, targetEnv)
+	if err != nil {
```

Comment:
Inside the err check if resp.statuscode is 404 then set d.SetID("") and return nil
else retrun err